src/program/latex/mod.rs

```diff
@@ -0,0 +1,1294 @@
+                            let mut close = None; // closing qubit in range
+
+                            // find the range between the qubits
+                            for (i, wire) in self.circuit.iter().enumerate() {
+                                // get each existing qubit in the circuit
+                                if *wire.0 == *ctrl || *wire.0 == *targ {
+                                    // if the qubit is the ctrl or target
+                                    if open.is_some() {
+                                        close = Some(i);
+                                        break;
+
+                                    // open qubit in range not found, set open qubit
+                                    } else {
+                                        open = Some(i)
+                                    }
+                                }
+                            }
+
+                            let mut vector: i64 = 0;
+                            if let Some(open) = open {
+                                if let Some(close) = close {
+                                    if ctrl < targ {
+                                        // a vector with a head from the ctrl to the targ
+                                        vector = (close as i64) - (open as i64);
+                                    } else {
+                                        // a vector with a head from the targ to the ctrl
+                                        vector = -((close as i64) - (open as i64));
+                                    }
+                                }
+                            }
+                            // set wire at column as the control qubit of target qubit
+                            // computed as the distance from the control qubit
+                            self.circuit
+                                .get_mut(ctrl)
+                                .and_then(|wire| wire.ctrl.insert(self.column, vector));
+                        }
+                    }
+                    // pred is None if relationship is split and no iterators are returned indicating this erroneous instruction
+                    if pred.is_none() {
+                        return Err(LatexGenError::FoundTargetWithNoControl);
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Analyzes a Gate from an Instruction and sets the gate at this column on
+    /// the wire. If the gate name is a composite gate, the gate name is
+    /// decomposed into canonical form before being set in the circuit. For
+    /// example, CNOT is a composite gate that can be decomposed into the
+    /// equivalent canonical form, CONTROLLED X.
+    ///
+    /// # Arguments
+    /// `self` - exposes all attributes in the diagram
+    /// `gate` - the Gate of the Instruction from `to_latex`.
+    fn parse_gate(&mut self, gate: &Gate) -> Result<(), LatexGenError> {
```

Comment:
Maybe more appropriately named `apply_gate`.

Reply:
That name actually sounds a bit more consuming to me. I think because there's a `set_gate` already. Also, this method is doing analysis on the gate, that's why I chose the name `parse_gate`. How would you say `apply_gate` is a better name for this method?

Reply:
Totally mixed up the words apply and insert after getting around to renaming this and changed the name to `insert_gate`. Maybe this name actually works better since I'm inserting the gate into the wire.

Reply:
Updated to `apply_gate`!